src/lib/call.cc

```diff
@@ -729,9 +729,156 @@ int Call::operator()(Call::argument_type &arg) {
                 }
             }
 
-            if(!calculate(read_depths, seq::char_index(ref_base), &stats)) {
+            size_t ref_index = seq::char_index(ref_base);
+            if(!calculate(read_depths, ref_index, &stats)) {
                 return;
             }
+
+            // Determine what nucleotides show up and the order they will appear in the REF and ALT field
+            // TODO: write tests that make sure REF="N" is properly handled
+            //      (1) N should be included in AD only if REF="N"
+            //      (2) N in AD should always be 0
+
+            // Measure total depth and sort nucleotides in descending order
+            typedef pair<int, int> key_t;
+            key_t total_depths[4] = {{0, 0}, {1, 0}, {2, 0}, {3, 0}};
+            int acgt_to_refalt_allele[5] = { -1, -1, -1, -1, -1}; // Maps allele to REF+ALT order
+            int refalt_to_acgt_allele[5] = { -1, -1, -1, -1, -1}; // Maps REF+ALT order to A,C,G,T,N order
+
+            int32_t dp_info = 0;
+            std::vector<int32_t> dp_counts(num_nodes, hts::bcf::int32_missing);
+            size_t dp_pos = library_start;
+            for(auto && a : read_depths) {
+                total_depths[0].second += a.counts[0];
+                total_depths[1].second += a.counts[1];
+                total_depths[2].second += a.counts[2];
+                total_depths[3].second += a.counts[3];
+                int32_t d = a.counts[0] + a.counts[1] + a.counts[2] + a.counts[3];
+                dp_info += d;
+                dp_counts[dp_pos++] = d;
+            }
+            sort(&total_depths[0], &total_depths[4], [](key_t a, key_t b) { return a.second > b.second; });
+
```

Comment:
You shouldn't need to recalculate total_depths as the AD tag can be copied from the input VCF line.